src/utils/useLocalStorage.ts

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+
+export const useLocalStorage = <T>(
+  key: string,
+  initialValue: T,
+) : [T, (value: T) => void] => {
+  const [datas, setDatas] = useState<T>(
+    JSON.parse(localStorage.getItem(key)!) || initialValue,
+  );
+
+  const saveDatas = (storageDatas: T) => {
```

Comment:
```suggestion
  const saveData = (storageData: T) => {
```
Data can't be plural